torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -943,9 +943,23 @@ struct to_ir {
       const SourceRange& val_range = subscript.value().range();
       Value* idx = emitExpr(subscript_exprs[0]);
       Value* val = sv->asValue(val_range, method);
-      auto node = graph->create(aten::Delete, {val, idx}, 0)
-                      ->setSourceRange(stmt.range());
-      graph->insertNode(node);
+
+      // If val is a class instance, this is a method call to a type-specific
+      // implementation of del defined in a __delitem__ method.
+      if (auto cls = val->type()->cast<ClassType>()) {
+        if (!cls->findMethod("__delitem__")) {
```

Comment:
If a method doesn't define `__delitem__`, we should just defer to the normal `environment_stack->removeVar(var.name(), /*check_if_removed=*/true);` behavior

Reply:
Hm, I thought this code path is for Subscript expressions (e.g. `del variable[key]`, where `variable` is a class, list or dict). If `variable` is a class but does not define `__delitem__`, this produces an error in Python too:
```
>>> class A:
...   def __init__(self):
...     pass
... 
>>> a = A()
>>> del a["key"]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'A' object does not support item deletion
```

Reply:
Oh word, yea i didn't realize this was within the subscript branch. Yea lgtm